enable the paywall landing page

cellog opened the pull request:
# Description

This PR implements 3 integration tests for the paywall. The first checks if a lock displays on the demo page, the second purchases a key, and the third checks to see if the unlock flag displays when a key exists on page reload.

<img width="1677" alt="Screen Shot 2019-03-16 at 1 00 45 PM" src="https://user-images.githubusercontent.com/98250/54481074-964ad100-4806-11e9-81bd-9c84911ede8a.png">
<img width="1677" alt="Screen Shot 2019-03-16 at 1 00 04 PM" src="https://user-images.githubusercontent.com/98250/54481075-964ad100-4806-11e9-8d7f-93a0b33a9dea.png">

# Issues

<!-- This PR should fix or reference at least one existing issue ID. Add or delete as appropriate. -->
Fixes #1882 
Fixes #1883 
Fixes #1885

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

<!--
PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)
-->

julien51 commented on paywall/src/pages/index.js:
```diff
@@ -1,13 +1,14 @@
-  )
+import Index from '../components/Index'
+
+export default function NextPaywall({ router }) {
```
Let's make sure default export have a name which matches the file's name.
I think `index.js` is not a great idea (given the fact that `index.js` files are treated differently for imports. Maybe `PaywallHome.js`?

And then you would name the component `PaywallHome`?

cellog replied:
again, for consistency, `index.js` is used for the page that is served by next in `unlock-app`, so I am not going to change that filename. `PaywallHome` is a fabulous name.

cellog replied:
fixed the name, PTAL

julien51 replied:
I really think it is important we keep the component names in sync with the file names so I submitted a PR that fixes this in unlock app: https://github.com/unlock-protocol/unlock/pull/2146
Please do the same thing here.

cellog replied:
roger. Done, PTAL

julien51 commented on paywall/src/pages/home.js:
```diff
@@ -0,0 +1,14 @@
+
+import Index from '../components/Index'
+
+export default function PaywallHome({ router }) {
```
So... I _still_ think the file should be named `paywalHome.js` so that we use the same name for the default export and the file name...

cellog replied:
done

cellog commented:
@julien51 I refactored the file naming and path structure to be consistent with `unlock-app`.

```
pages/
  paywallHome.js (component inside is PaywallHome)
components/
  content/
(this is the page content for PaywallHome, and does the switching between app and landing page)
    PaywallHomeContent.js
(this is the paywall app)
    PaywallAppContent.js
(this is the landing page)
    PaywallLandingPage.js
```

The tests and stories are also refactored to reflect both the path and the naming structure. PTAL when you get a chance.

julien51 commented on paywall/next.config.js:
```diff
@@ -44,7 +44,7 @@ module.exports = withSourceMaps({
 
     return {
-      '/': { page: '/' },
+      '/': { page: '/paywallHome' },
```
I am not sure we need to prefix all pages with `paywall` since this is the paywall app, this is pretty obvious right?

cellog replied:
I literally did this because you asked me to ;). `'/'` refers to `src/pages/index.js` and that is why I was using it originally. We renamed the page from `index.js` and that forces updating both here and in `_server.js`.

julien51 replied:
Sorry about the confusion here. I think what we need is `'/': { page: '/home' },`!

julien51 commented on package.json:
```diff
@@ -14,16 +14,17 @@
     "eslint-plugin-jsx-a11y": "^6.2.1",
     "eslint-plugin-mocha": "^5.3.0",
     "eslint-plugin-node": "^8.0.1",
+    "eslint-plugin-prettier": "^3.0.1",
```
Please, open a separate PR for the deps please?

cellog replied:
this is just missing a rebase. These deps are already in master

julien51 commented on paywall/src/_server.js:
```diff
@@ -19,9 +19,9 @@ function _server(port, dev) {
+          if (pathname.match('/0x') || pathname === '/') {
             const params = route('/:lockAddress/:redirect?')(pathname)
-            app.render(req, res, '/', Object.assign(params, query))
+            app.render(req, res, '/paywallHome', Object.assign(params, query))
```
This is a bit confusing. Does this mean the home page is at `/paywallHome`?

cellog replied:
see above.

julien51 replied:
Well this is exactly what I was saying i my PR then ;) Next is too strict.
Of course we want the home page to be available on `/` and not expect the user to go to `/paywallHome`.

cellog commented on paywall/src/_server.js:
```diff
@@ -19,9 +19,9 @@ function _server(port, dev) {
           // assigning `query` into the params means that we still
           // get the query string passed to our application
-          if (pathname.match('/0x')) {
+          if (pathname.match('/0x') || pathname === '/') {
```
If I were to split the PR, this is literally the only change that would be in the 2nd PR.

julien51 replied:
I know... but it is still cleaner? also very easy to actually do with git?

julien51 commented:
Well, in `unlock-app`, the home page content component is is just at called `HomeContent`, not `UnlockAppHomeContent`.... I think we should have the same thing here.

cellog commented:
please take a look, @julien51 The changes you requested are in:

 - line enabling the landing page no longer enables it.
 - paywallHome -> home
